src/main/groovy/io/github/cosmicsilence/scalafix/ScalafixTask.groovy

```diff
@@ -29,46 +30,41 @@ class ScalafixTask extends SourceTask {
     ScalafixMainMode mode
 
     @Input
-    @Optional
-    String scalaVersion
+    final Property<String> scalaVersion = project.objects.property(String)
 
     @Input
     @Optional
-    List<String> compileOptions
+    final ListProperty<String> compileOptions = project.objects.listProperty(String)
 
     @Input
     @Optional
-    List<String> classpath
+    final ListProperty<String> classpath = project.objects.listProperty(String)
 
     @Input
     String sourceRoot
 
-    @Input
+    @Internal
     Boolean semanticdbConfigured
 
     @TaskAction
     void run() {
-        if (!source.isEmpty()) processSources()
-        else logger.warn("No sources to be processed")
```

Comment:
this check was unnecessary. The task is automatically skipped when `source` is empty as it is annotated with `@SkipWhenEmpty` on the base class